# Partial backport for kubev2v/forklift v2.12.9.
# Applies to pkg/controller/plan/adapter/vsphere/builder_test.go from:
# https://github.com/kubev2v/forklift/tree/v2.12.9
#
# The release branch contains the trunc-15 implementation but retained the
# earlier trunc-4 assertion. Backport only the matching assertion/import hunk
# from upstream commit 48fe928f9bf196f3a20ec09ad6d41937b728b258:
# https://github.com/kubev2v/forklift/commit/48fe928f9bf196f3a20ec09ad6d41937b728b258
--- a/pkg/controller/plan/adapter/vsphere/builder_test.go
+++ b/pkg/controller/plan/adapter/vsphere/builder_test.go
@@ -2,7 +2,6 @@ package vsphere
 
 import (
 	"context"
-	"fmt"
 
 	v1beta1 "github.com/kubev2v/forklift/pkg/apis/forklift/v1beta1"
 	"github.com/kubev2v/forklift/pkg/apis/forklift/v1beta1/plan"
@@ -247,8 +246,9 @@ var _ = Describe("vSphere builder", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(pvcs).To(HaveLen(1))
 			pvc := pvcs[0]
-			// The default template now uses trunc 4 for both plan and VM names
-			Expect(pvc.Name).Should(HavePrefix(fmt.Sprintf("%.4s-%.4s-disk-", builder.Plan.Name, vm.Name)))
+			// Default template: trunc(15,"unit-test-plan-single-vm") = "unit-test-plan-"
+			// + "-" + trunc(15,"customer-frontend-server") = "customer-fronte" + "-disk-"
+			Expect(pvc.Name).Should(HavePrefix("unit-test-plan--customer-fronte-disk-"))
 			Expect(pvc.Spec.DataSourceRef.Kind).To(Equal(v1beta1.VSphereXcopyVolumePopulatorKind))
 			Expect(pvc.Spec.DataSourceRef.APIGroup).To(Equal(&v1beta1.SchemeGroupVersion.Group))
 			Expect(pvc.Spec.DataSourceRef.Name).To(Equal(pvc.Name))
